indico/modules/events/registration/clone.py

```diff
@@ -22,48 +22,85 @@
             db.session.add(new_form)
             db.session.flush()
 
-    def _find_registration_forms(self):
-        return RegistrationForm.find(~RegistrationForm.is_deleted,
-                                     RegistrationForm.event_id == int(self.event.id))
-
     def _clone_form_items(self, old_form, new_form):
         old_sections = RegistrationFormSection.find(RegistrationFormSection.registration_form_id == old_form.id,
                                                     ~RegistrationFormSection.is_deleted)
-        section_attrs = get_simple_column_attrs(RegistrationFormSection) - {'registration_form_id'}
+        items_attrs = get_simple_column_attrs(RegistrationFormSection)
         for old_section in old_sections:
             new_section = RegistrationFormSection(registration_form=new_form, **{attr: getattr(old_section, attr)
-                                                                                 for attr in section_attrs})
+                                                                                 for attr in items_attrs})
             old_section_items = RegistrationFormItem.find(~RegistrationFormItem.is_deleted, parent=old_section)
-            items_attrs = get_simple_column_attrs(RegistrationFormItem) | {'parent_id'}
             for old_item in old_section_items:
                 new_item = RegistrationFormItem(parent=new_section, registration_form=new_form,
                                                 **{attr: getattr(old_item, attr) for attr in items_attrs})
                 if new_item.is_field:
-                    field_data = RegistrationFormFieldData(field=new_item,
-                                                           versioned_data=getattr(old_item, 'versioned_data'))
-                    new_item.current_data = field_data
+                    if self._clone_with_registrations:
+                        self._clone_all_field_versions(old_item, new_item)
+                    else:
+                        field_data = RegistrationFormFieldData(field=new_item,
+                                                               versioned_data=getattr(old_item, 'versioned_data'))
+                        new_item.current_data = field_data
                 new_section.children.append(new_item)
             db.session.add(new_section)
             db.session.flush()
+
+    def _clone_all_field_versions(self, old_item, new_item):
+        for old_version in old_item.data_versions:
+            new_version = RegistrationFormFieldData(versioned_data=old_version.versioned_data, field=new_item)
+            if old_version.id == old_item.current_data_id:
+                new_item.current_data = new_version
+            new_item.data_versions.append(new_version)
+            self._field_data_dict[old_version.id] = new_version
+
+    def _clone_registrations(self, old_form, new_form, new_event):
+        registration_attrs = get_simple_column_attrs(Registration) - {'base_price', 'uuid', 'ticket_uuid'}
```

Comment:
We are not copying the payment module settings. We said before that we don't want users to be able to set `base_price` in case payment module is not enabled.